tests/e2e/lib/apps.go

```diff
@@ -108,6 +132,112 @@ func UninstallApplication(ocClient client.Client, file string) error {
 // 		return true, err
 // 	}
 // }
+func HasDCsInNamespace(ocClient client.Client, namespace string) (bool, error) {
+	dcList := &ocpappsv1.DeploymentConfigList{}
+	err := ocClient.List(context.Background(), dcList, client.InNamespace(namespace))
+	if err != nil {
+		return false, err
+	}
+	if len(dcList.Items) == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+func HasReplicationControllersInNamespace(ocClient client.Client, namespace string) (bool, error) {
+	rcList := &corev1.ReplicationControllerList{}
+	err := ocClient.List(context.Background(), rcList, client.InNamespace(namespace))
+	if err != nil {
+		return false, err
+	}
+	if len(rcList.Items) == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+func HasTemplateInstancesInNamespace(ocClient client.Client, namespace string) (bool, error) {
+	tiList := &templatev1.TemplateInstanceList{}
+	err := ocClient.List(context.Background(), tiList, client.InNamespace(namespace))
+	if err != nil {
+		// if runtime.IsNotRegisteredError(err) {
+		// 	return false, nil
+		// }
+		return false, err
+	}
+	if len(tiList.Items) == 0 {
+		return false, nil
+	}
+	return true, nil
+}
+
+func NamespaceRequiresResticDCWorkaround(ocClient client.Client, namespace string) (bool, error) {
+	hasDC, err := HasDCsInNamespace(ocClient, namespace)
+	if err != nil {
+		return false, err
+	}
+	hasRC, err := HasReplicationControllersInNamespace(ocClient, namespace)
+	if err != nil {
+		return false, err
+	}
+	hasTI, err := HasTemplateInstancesInNamespace(ocClient, namespace)
+	if err != nil {
+		return false, err
+	}
+	return hasDC || hasRC || hasTI, nil
+}
+
+func AreVolumeSnapshotsReady(ocClient client.Client, backupName string) wait.ConditionFunc {
+	return func() (bool, error) {
+		vList := &volumesnapshotv1.VolumeSnapshotList{}
+		// vListBeta := &volumesnapshotv1beta1.VolumeSnapshotList{}
+		err := ocClient.List(context.Background(), vList, &client.ListOptions{LabelSelector: label.NewSelectorForBackup(backupName)})
+		if err != nil {
+			// try beta version
+			// if runtime.IsNotRegisteredError(err) {
+			// 	// try v1beta1
+			// 	vList = nil // reset
+			// 	err = ocClient.List(context.Background(), vList, client.InNamespace(namespace))
+			// 	if err != nil {
+			// 		if runtime.IsNotRegisteredError(err) {
+			// 			return false, nil
+			// 		}
+			// 	}
+			// }
```

Comment:
Do we want to keep this commented code ?

Reply:
resolved @deepakraj1997 @shubham-pampattiwar